src/IB/FEMechanicsExplicitIntegrator.cpp

```diff
@@ -0,0 +1,647 @@
+{
+    FEMechanicsBase::doInitializeFEEquationSystems();
+    d_system_dof_map_cache.resize(d_meshes.size());
+    const bool from_restart = RestartManager::getManager()->isFromRestart();
+    for (unsigned int part = 0; part < d_meshes.size(); ++part)
+    {
+        EquationSystems& equation_systems = *d_equation_systems[part];
+        IBTK::setup_system_vectors(&equation_systems,
+                                   { COORDS_SYSTEM_NAME, VELOCITY_SYSTEM_NAME },
+                                   { "current", "half", "new" },
+                                   from_restart);
+        IBTK::setup_system_vectors(
+            &equation_systems, { FORCE_SYSTEM_NAME }, { "current", "half", "new", "tmp", "RHS Vector" }, from_restart);
+    }
+}
+
+void
+FEMechanicsExplicitIntegrator::doInitializeFESystemVectors()
+{
+    FEMechanicsBase::doInitializeFESystemVectors();
+    std::vector<EquationSystems*> equation_systems;
+    for (const auto& es : d_equation_systems) equation_systems.push_back(es.get());
+    d_X_vecs.reset(new LibMeshSystemVectors(equation_systems, COORDS_SYSTEM_NAME));
+    d_U_vecs.reset(new LibMeshSystemVectors(equation_systems, VELOCITY_SYSTEM_NAME));
+    d_F_vecs.reset(new LibMeshSystemVectors(equation_systems, FORCE_SYSTEM_NAME));
+    if (d_has_static_pressure_parts)
+    {
+        d_P_vecs.reset(new LibMeshSystemVectors(equation_systems, d_static_pressure_part, PRESSURE_SYSTEM_NAME));
+    }
+    if (d_has_dynamic_pressure_parts)
+    {
+        d_P_vecs.reset(new LibMeshSystemVectors(equation_systems, d_dynamic_pressure_part, PRESSURE_SYSTEM_NAME));
+    }
+}
+
+void
+FEMechanicsExplicitIntegrator::doInitializeFEData(const bool use_present_data)
+{
+    FEMechanicsBase::doInitializeFEData(use_present_data);
+}
+
+FEData::SystemDofMapCache*
+FEMechanicsExplicitIntegrator::getDofMapCache(const std::string& system_name, const unsigned int part)
+{
+    return getDofMapCache(d_equation_systems[part]->get_system(system_name).number(), part);
+}
+
+FEData::SystemDofMapCache*
+FEMechanicsExplicitIntegrator::getDofMapCache(unsigned int system_num, const unsigned int part)
+{
+    System& system = d_equation_systems[part]->get_system(system_num);
+    const unsigned int n_vars = system.n_vars();
+    TBOX_ASSERT(n_vars > 0);
+    const FEType& fe_type = system.variable_type(0);
+
+    // we assume that all variables have the same FEType inside IBAMR
+    for (unsigned int var_n = 1; var_n < n_vars; ++var_n)
+    {
+        TBOX_ASSERT(fe_type == system.variable_type(var_n));
+    }
+
+    // We assume that the dofs have the same layout as long as they have the
+    // same fe type and number of variables
+    std::pair<unsigned int, FEType> key(n_vars, fe_type);
+
+    std::unique_ptr<FEData::SystemDofMapCache>& dof_map_cache = d_system_dof_map_cache[part][key];
+    if (dof_map_cache == nullptr)
+    {
+        dof_map_cache.reset(new FEData::SystemDofMapCache(system));
+    }
+    return dof_map_cache.get();
```

Comment:
What's up with this function? Is there really no way to share this information (i.e., use the common `FEData` object that contains these things)?

Reply:
Good question. It does not seem to be used; I will remove it.